pkgs/by-name/op/opencode/package.nix

```
@@ -1,60 +1,147 @@
+
+    subPackages = [ "cmd/opencode" ];
+
+    env.CGO_ENABLED = 0;
+
+    ldflags = [
+      "-s"
+      "-w"
+      "-X=main.Version=${version}"
+    ];
+  };
+
+  opencode-node-modules-hash = {
+    "aarch64-darwin" = "sha256-5Y9bfzYaO1iwSgpkCwPuUCDRtG8uXwgKm98sq78HisI=";
+    "aarch64-linux" = "sha256-t7dcyPsWeqYJv4/DPP15bvVurVPGCXWXVPUfpk1+l1Q=";
+    "x86_64-darwin" = "sha256-qYljdz4iyl3aZBhVU+uIVx4ZsvY9ubTTLNxJ94TUkBo=";
+    "x86_64-linux" = "sha256-TYjcA7MTfOKkvTwGSK1SotMDIH9xag79dikf6hMF8us=";
+  };
+  node_modules = stdenv.mkDerivation {
+    name = "opencode-${version}-node-modules";
+    inherit version src;
+
+    impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
+      "GIT_PROXY_COMMAND"
+      "SOCKS_SERVER"
+    ];
+
+    nativeBuildInputs = [ bun ];
+
+    dontConfigure = true;
+
+    buildPhase = ''
+      bun install --no-cache --no-progress --frozen-lockfile  --filter=opencode
+    '';
+
+    installPhase = ''
+      mkdir -p $out/node_modules
+
+      cp -R ./node_modules $out
+    '';
+
+    # Required else we get errors that our fixed-output derivation references store paths
+    dontFixup = true;
+
+    outputHash = opencode-node-modules-hash.${stdenv.hostPlatform.system};
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
   };
 
-  vendorHash = "sha256-pnev0o2/jirTqG67amCeI49XUdMCCulpGq/jYqGqzRY=";
+  modelsDevData = fetchurl {
+    url = "https://models.dev/api.json";
+    sha256 = "sha256-+VpQjvOOtuoltE3n9MZn8d0srYDacOh6B4wM0RqBmBM=";
+  };
+  bun-target = {
+    "aarch64-darwin" = "bun-darwin-arm64";
+    "aarch64-linux" = "bun-linux-arm64";
+    "x86_64-darwin" = "bun-darwin-x64";
+    "x86_64-linux" = "bun-linux-x64";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opencode";
+  inherit version src;
+
+  nativeBuildInputs = [ bun ];
+
+  patches = [ ./fix-models-macro.patch ];
 
-  ldflags = [
-    "-s"
-    "-w"
-    "-X github.com/sst/opencode/internal/version.Version=${finalAttrs.version}"
-  ];
+  configurePhase = ''
+    runHook preConfigure
 
-  checkFlags =
-    let
-      skippedTests = [
-        # permission denied
-        "TestBashTool_Run"
-        "TestSourcegraphTool_Run"
-        "TestLsTool_Run"
+    cp -R ${node_modules}/node_modules .
 
-        # Difference with snapshot
-        "TestGetContextFromPaths"
-      ];
-    in
-    [ "-skip=^${lib.concatStringsSep "$|^" skippedTests}$" ];
+    runHook postConfigure
+  '';
 
-  nativeCheckInputs = [
-    versionCheckHook
-  ];
-  versionCheckProgramArg = "--version";
-  doInstallCheck = true;
+  buildPhase = ''
+    runHook preBuild
 
-  passthru.updateScript = nix-update-script { };
+    export MODELS_JSON="$(cat ${modelsDevData})"
+    bun build --define OPENCODE_VERSION="'${version}'" --compile --minify --target=${
+      bun-target.${stdenv.hostPlatform.system}
+    } --outfile=opencode ./packages/opencode/src/index.ts ${opencode-tui}/bin/opencode
```

```suggestion
    bun build \
      --define OPENCODE_VERSION="'${version}'" \
      --compile \
      --minify \
      --target=${bun-target.${stdenv.hostPlatform.system}} \
      --outfile=opencode
      ./packages/opencode/src/index.ts ${opencode-tui}/bin/opencode
```

Nit (can be ignored): This would be more readable.